codex/sales/reservations.nim

```diff
@@ -370,54 +398,61 @@ proc createAvailability*(
 
   return success(availability)
 
-proc createReservation*(
+method createReservation*(
   self: Reservations,
   availabilityId: AvailabilityId,
   slotSize: UInt256,
   requestId: RequestId,
   slotIndex: UInt256
-): Future[?!Reservation] {.async.} =
+): Future[?!Reservation] {.async, base.} =
 
-  trace "creating reservation", availabilityId, slotSize, requestId, slotIndex
+  withLock(self.availabilityLock):
+    without availabilityKey =? availabilityId.key, error:
+      return failure(error)
 
-  let reservation = Reservation.init(availabilityId, slotSize, requestId, slotIndex)
+    without availability =? await self.get(availabilityKey, Availability), error:
+      return failure(error)
 
-  without availabilityKey =? availabilityId.key, error:
-    return failure(error)
+    # We recheck that the found availability is still matching after we have lock acquired
+    if availability.freeSize < slotSize:
+       # Lets monitor how often this happen and if it is often we can make it more inteligent to handle it
+      codex_reservations_availability_mismatch.inc()
+      warn "we matched availability but then it changed and now it does not match anymore", id = availability.id
```

Comment:
Why is this a warning? I don't think it's bad if the lock works as intended. The log should also be logged at the callsite as we don't know where it came from (technically). 

And don't forget to capitalize the start 😝

Reply:
Okey, lets keep the metric then. I want it in order to see how often this will be happening as we could optimize this further in the future...